woocommerce/woocommerce
# [CYS] Proceed with product creation when their images fail to upload

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR avoids returning an error when the `product_image_upload` function fails. The error is now handled in the `product_update` function before associating the image to the product. That way we ensure products are created even when the image upload fails.

Closes https://github.com/woocommerce/woocommerce/issues/44005



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Head over to `/wp-admin/tools.php?page=woocommerce-admin-test-helper` and enable `customize-store` feature flag.
2. Go to `WooCommerce > Home`, click on the `Start Customizing` button, and finish the CYS flow.
3. Make sure all products are created correctly and have images associated with them.
4. Update the code in https://github.com/woocommerce/woocommerce/blob/97dca87e7ae2f24a10be3ac9ddc27598952764ab/plugins/woocommerce/src/Blocks/AIContent/UpdateProducts.php#L305 👇 to simulate an error when uploading product images:
```
	private function product_image_upload( $product_id, $image_src, $image_alt ) {
		return new \WP_Error( 'error_uploading_image' );
```
5. Go to `wp-admin/tools.php?page=woocommerce-admin-test-helper` and click on `Reset Customize Your Store` and `Delete all products`.
6. Go to `WooCommerce > Home`, click on the `Start Customizing` button, and finish the CYS flow.
7. Go to `Products` in the admin and make sure all dummy products are created with a placeholder image.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

CYS: Proceed with product creation when their images fail to upload.

#### Comment 

</details>
